app/javascript/components/key-performance-indicators/components/date-range-select/container.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { Select, MenuItem, FormControl } from "@material-ui/core";
+
+import DateRangeDialog from "../date-range-dialog";
+
+import DateRange from "../../date-range";
+
+const DateRangeSelect = ({
+  ranges,
+  selectedRange,
+  withCustomRange,
+  setSelectedRange,
+  disabled
+}) => {
+  // FIXME: We should use 'useI18n' but it returns a null object when called
+  //        from here.
+  const i18n = window.I18n;
+
+  const [showRangePicker, setShowRangePicker] = useState(false);
+
+  const [customRange, setCustomRange] = useState(
+    new DateRange(
+      "custom-range",
```

Comment:
In a seperate constants.js file, add custom-range and Custom Range. Also should this use i18n?

Reply:
Overall we should use const for repeated strings